Assets/MixedRealityToolkit.Tests/EditModeTests/Core/TestFixture_02_MixedRealityToolkitFilesTests.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Utilities.Editor;
+using NUnit.Framework;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+using UnityEngine.TestTools;
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.MixedReality.Toolkit.Tests.Core
+{
+    // Tests for the MixedRealityToolkitFiles utility class
+    public class TestFixture_02_MixedRealityToolkitFilesTests
+    {
+        [Test]
+        public void Test_01_FindMatchingModule()
+        {
+            TestInvalidPath("");
+
+            // Test invalid base name
+            TestInvalidPath("aaa");
+            TestInvalidPath(".SDK");
+            TestInvalidPath("aaa.SDK");
+
+            // Test missing chars
+            TestInvalidPath("MixedRealityToolki.SDK");
+            TestInvalidPath("MixedRealityToolkit.SD");
+            TestInvalidPath("ixedRealityToolkit.SDK");
+            TestInvalidPath("MixedRealityToolkit.DK");
+
+            // Test missing dots
+            TestInvalidPath("SDK");
+            TestInvalidPath("MixedRealityToolkitSDK");
+
+            // Test valid paths
+            TestValidPath("MixedRealityToolkit", MixedRealityToolkitModuleType.Core);
```

Comment:
In the real code, is it always getting relative paths, or is it getting absolute paths i.e.:

c:\path\to\project\and\mrtk
/path/to/project/and/mrtk

If it's relative then this is good, if there are absolute paths involved it would be good to get those cases in as well, as those are more like the "real" stuff